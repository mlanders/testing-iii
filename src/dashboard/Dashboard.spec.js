import React from 'react';
// import { render, fireEvent } from 'react-testing-library';
import 'jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import 'react-testing-library/cleanup-after-each';

import Dashboard from './Dashboard';
import { render } from 'react-testing-library';

describe('Dashboard Component', () => {
	test('matches snapshot', () => {
		const tree = renderer.create(<Dashboard />);
		expect(tree.toJSON()).toMatchSnapshot();
	});
	test('component does not crash', () => {
		render(<Dashboard />);
	});
});
