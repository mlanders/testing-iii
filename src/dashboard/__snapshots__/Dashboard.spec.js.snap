// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component matches snapshot 1`] = `
Array [
  <div
    className="display panel"
  >
    <div
      className="led green-led"
      data-testid="lock"
    >
      Unlocked
    </div>
    <div
      className="led green-led"
      data-testid="close"
    >
      Open
    </div>
  </div>,
  <div
    className="controls panel"
  >
    <button
      className="toggle-btn"
      data-testid="lockBtn"
      disabled={true}
      onClick={[Function]}
    >
      Lock Gate
    </button>
    <button
      className="toggle-btn"
      data-testid="closeBtn"
      disabled={false}
      onClick={[Function]}
    >
      Close Gate
    </button>
  </div>,
]
`;
